### PR TITLE
Migrate to plain CSS (remove CSS modules)

### LINKED ISSUE #418: Migrate to plain (post-processed) CSS
The CMS has a number of conflicting CSS architectures in place. Up until now we've loosely expected to settle into CSS modules. The problem with CSS modules is they're still just static CSS, and they still have to live in their own file (ideally per-component).

[Styled components](https://styled-components.com/) is getting pretty mature, and would allow us to house (dynamic!) styles right in our component files. It supports theming, too.

Let's discuss!

---

## Update

Plain ol' CSS is really seeming like the right approach. This ticket is being updated to cover migration to plain CSS (with cssnext for future syntax support). Note that multiple contributors can collaborate to make this happen.

### PR DESCRIPTION
**Pull request by Benaiah**
**- Summary**

Migrate to plain CSS and remove CSS modules. Closes #418.

**- Test plan**

Manually tested visual elements; I would _really_ appreciate help checking all the nooks and crannies of the UI. Most CSS should be exactly equivalent but there's always the chance I missed something.

**- Description for the changelog**

Migrate to plain CSS and remove CSS modules.

**- A picture of a cute animal (not mandatory but encouraged)**

image

### CODE REVIEW

File: src/lib/styleHelper.js [hunk 1]
```diff
@@ -0,0 +1 @@
+export const prefixer = prefix => className => `ncms-${ prefix }-${ className }`;
```
Comment (dopry) [inline src/lib/styleHelper.js]: We should document the prefixer. 

Is there a runtime impact of this implementation? Is my browser going to be building these strings every time it renders a component? If we're writing the style name in the css file, why not use a static name in the templates?

**Comment by tech4him1** [comment]
@Benaiah Did we decide to explicitly write `ncms-` in all the CMS files? or should we just add that when we run PostCSS?

**Comment by Benaiah** [comment]
@tech4him1 we haven’t found any good way to do that, other than remaining on CSS modules. We have some global class styles as well for various reasons, so simply prefixing every class wouldn’t work, even if I knew of a tool that did that for us

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by erquhart** [comment]
@Benaiah I'm thinking:

1. Use `nc` as the prefix.
2. Drop the prefixer.

**Comment by Benaiah** [comment]
@erquhart sounds good to me; I'll try to get that done & pushed today.

**Comment by Benaiah** [comment]
Note that this is currently WIP - while the styles _should_ be working, I may need to clean up the output of the script I used to replace prefixer calls, and I still need to actually remove the prefixer and the lines which import it.

File: src/components/Widgets/FileControl.css [hunk 2]
```diff
@@ -1,16 +1,16 @@
 @import "../UI/theme.css";
 
-.input {
+.nc-fileControlinput {
```
Comment (dopry) [inline src/components/Widgets/FileControl.css]: missing - after fileControl

File: src/components/stories/Card.js [hunk 3]
```diff
@@ -37,6 +37,6 @@ storiesOf('Card', module)
       <img src="http://www.top13.net/wp-content/uploads/2015/10/perfectly-timed-funny-cat-pictures-5.jpg" />
       <h1>Now with footer.</h1>
       <p>header and footer elements are also not subject to margin</p>
-      <footer style={styles.footer}>&copy; Thousand Cats Corp</footer>
+      <footer style={styles("footer")}>&copy; Thousand Cats Corp</footer>
```
Comment (dopry) [inline src/components/stories/Card.js]: should this be a class in place of style?

Comment (Benaiah) [reply src/components/stories/Card.js]: Actually, this should stay the way it was before, and it got changed to this when changing the prefixer to a function. In this case the `styles` object is a manually constructed object earlier in the file, and has nothing to do with the prefixer or CSS modules. I'll change this back.

File: src/components/Widgets/FileControl.css (hunk 2 above)
Comment (Benaiah) [reply src/components/Widgets/FileControl.css]: Thanks, I'll get this fixed.

File: src/components/ControlPanel/ControlPane.js [hunk 4]
```diff
@@ -31,12 +31,12 @@ export default class ControlPane extends Component {
     const value = entry.getIn(['data', fieldName]);
     const metadata = fieldsMetaData.get(fieldName);
     const errors = fieldsErrors.get(fieldName);
-    const labelClass = errors ? styles.labelWithError : styles.label;
+    const labelClass = errors ? `nc-controlPane-label nc-controlPane-labelWithError` : 'nc-controlPane-label';
```
Comment (Benaiah) [inline src/components/ControlPanel/ControlPane.js]: Quotes here should be fixed.

File: src/components/EntryEditor/EntryEditor.js [hunk 5]
```diff
@@ -9,9 +9,10 @@ import ControlPane from '../ControlPanel/ControlPane';
 import PreviewPane from '../PreviewPane/PreviewPane';
 import Toolbar from './EntryEditorToolbar';
 import { StickyContext } from '../UI/Sticky/Sticky';
-import styles from './EntryEditor.css';
 import stickyStyles from '../UI/Sticky/Sticky.css';
 
+
+
```
Comment (Benaiah) [inline src/components/EntryEditor/EntryEditor.js]: Remove extra blank lines from edited JS files.

File: src/components/ControlPanel/ControlPane.js [hunk 6]
```diff
@@ -4,7 +4,7 @@ import { Map, fromJS } from 'immutable';
 import ImmutablePropTypes from 'react-immutable-proptypes';
 import { resolveWidget } from '../Widgets';
 import ControlHOC from '../Widgets/ControlHOC';
-import styles from './ControlPane.css';
+
```
Comment (Benaiah) [inline src/components/ControlPanel/ControlPane.js]: Remove extra blank lines from edited JS files.

File: src/components/AppHeader/AppHeader.js [hunk 7]
```diff
@@ -8,7 +8,7 @@ import AppBar from "react-toolbox/lib/app_bar";
 import FontIcon from "react-toolbox/lib/font_icon";
 import FindBar from "../FindBar/FindBar";
 import { stringToRGB } from "../../lib/textHelper";
-import styles from "./AppHeader.css";
+
```
Comment (Benaiah) [inline src/components/AppHeader/AppHeader.js]: Remove extra blank lines from edited JS files.

File: src/backends/test-repo/AuthenticationPage.js [hunk 8]
```diff
@@ -4,7 +4,7 @@ import Input from "react-toolbox/lib/input";
 import Button from "react-toolbox/lib/button";
 import { Card, Icon } from "../../components/UI";
 import logo from "../git-gateway/netlify_logo.svg";
-import styles from "../git-gateway/AuthenticationPage.css";
+
```
Comment (Benaiah) [inline src/backends/test-repo/AuthenticationPage.js]: Remove extra blank lines from edited JS files.

File: src/backends/github/AuthenticationPage.js [hunk 9]
```diff
@@ -5,7 +5,7 @@ import Authenticator from '../../lib/netlify-auth';
 import { Icon } from '../../components/UI';
 import { Notifs } from 'redux-notifications';
 import { Toast } from '../../components/UI/index';
-import styles from './AuthenticationPage.css';
+
```
Comment (Benaiah) [inline src/backends/github/AuthenticationPage.js]: Remove extra blank lines from edited JS files.

File: src/backends/git-gateway/AuthenticationPage.js [hunk 10]
```diff
@@ -6,7 +6,7 @@ import { Notifs } from 'redux-notifications';
 import { Toast } from '../../components/UI/index';
 import { Card, Icon } from "../../components/UI";
 import logo from "./netlify_logo.svg";
-import styles from "./AuthenticationPage.css";
+
```
Comment (Benaiah) [inline src/backends/git-gateway/AuthenticationPage.js]: Remove extra blank lines from edited JS files.

File: src/components/EntryListing/EntryListing.js [hunk 11]
```diff
@@ -7,7 +7,7 @@ import history from '../../routing/history';
 import { resolvePath } from '../../lib/pathHelper';
 import { selectFields, selectInferedField } from '../../reducers/collections';
 import { Card } from '../UI';
-import styles from './EntryListing.css';
+
```
Comment (Benaiah) [inline src/components/EntryListing/EntryListing.js]: Remove extra blank lines from edited JS files.

File: src/components/FindBar/FindBar.js [hunk 12]
```diff
@@ -1,6 +1,6 @@
 import PropTypes from 'prop-types';
 import React, { Component } from 'react';
-import styles from './FindBar.css';
+
```
Comment (Benaiah) [inline src/components/FindBar/FindBar.js]: Remove extra blank lines from edited JS files.

File: src/components/PreviewPane/PreviewPane.js [hunk 13]
```diff
@@ -10,7 +10,7 @@ import { INFERABLE_FIELDS } from '../../constants/fieldInference';
 import PreviewContent from './PreviewContent.js';
 import PreviewHOC from '../Widgets/PreviewHOC';
 import Preview from './Preview';
-import styles from './PreviewPane.css';
+
```
Comment (Benaiah) [inline src/components/PreviewPane/PreviewPane.js]: Remove extra blank lines from edited JS files.

File: src/components/UI/Sticky/Sticky.js [hunk 14]
```diff
@@ -2,7 +2,7 @@ import PropTypes from 'prop-types';
 import React, { Component } from 'react';
 import classnames from 'classnames';
 import { partial, without } from 'lodash';
-import styles from './Sticky.css';
+
```
Comment (Benaiah) [inline src/components/UI/Sticky/Sticky.js]: Remove extra blank lines from edited JS files.

File: src/components/UI/icon/Icon.js [hunk 15]
```diff
@@ -1,5 +1,5 @@
 import React from 'react';
-import styles from './Icon.css';
+
```
Comment (Benaiah) [inline src/components/UI/icon/Icon.js]: Remove extra blank lines from edited JS files.

File: src/components/UI/loader/Loader.js [hunk 16]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
 import CSSTransition from 'react-transition-group/CSSTransition';
-import styles from './Loader.css';
+
```
Comment (Benaiah) [inline src/components/UI/loader/Loader.js]: Remove extra blank lines from edited JS files.

File: src/components/UnpublishedListing/UnpublishedListing.js [hunk 17]
```diff
@@ -9,7 +9,7 @@ import { Card, CardTitle, CardText, CardActions } from 'react-toolbox/lib/card';
 import Button from 'react-toolbox/lib/button';
 import UnpublishedListingCardMeta from './UnpublishedListingCardMeta.js';
 import { status, statusDescriptions } from '../../constants/publishModes';
-import styles from './UnpublishedListing.css';
+
```
Comment (Benaiah) [inline src/components/UnpublishedListing/UnpublishedListing.js]: Remove extra blank lines from edited JS files.

File: src/components/UnpublishedListing/UnpublishedListingCardMeta.js [hunk 18]
```diff
@@ -1,12 +1,12 @@
 import PropTypes from 'prop-types';
 import React from 'react';
-import styles from './UnpublishedListingCardMeta.css';
+
```
Comment (Benaiah) [inline src/components/UnpublishedListing/UnpublishedListingCardMeta.js]: Remove extra blank lines from edited JS files.

File: src/components/Widgets/BooleanControl.js [hunk 19]
```diff
@@ -3,15 +3,15 @@ import React from 'react';
 import ImmutablePropTypes from "react-immutable-proptypes";
 import Switch from 'react-toolbox/lib/switch';
 import { isBoolean } from 'lodash';
-import styles from './BooleanControl.css';
+
```
Comment (Benaiah) [inline src/components/Widgets/BooleanControl.js]: Remove extra blank lines from edited JS files.

File: src/components/Widgets/FileControl.css [hunk 20]
```diff
@@ -1,16 +1,16 @@
...
   display: none !important;
 }
 
-.message {
+.nc-fileControlmessage {
```
Comment (Benaiah) [inline src/components/Widgets/FileControl.css]: Fix bad identifier.

File: src/components/Widgets/FileControl.css [hunk 21]
```diff
@@ -1,16 +1,16 @@
...
   padding: 20px;
   display: block;
   font-size: 12px;
 }
 
-.imageUpload {
+.nc-fileControlimageUpload {
```
Comment (Benaiah) [inline src/components/Widgets/FileControl.css]: Fix bad identifier.

File: src/components/Widgets/FileControl.js [hunk 22]
```diff
@@ -3,7 +3,7 @@ import React from 'react';
 import { truncateMiddle } from '../../lib/textHelper';
 import { Loader } from '../UI';
 import AssetProxy, { createAssetProxy } from '../../valueObjects/AssetProxy';
-import styles from './FileControl.css';
+
```
Comment (Benaiah) [inline src/components/Widgets/FileControl.js]: Remove extra blank lines from edited JS files.

File: src/components/Widgets/ImageControl.js [hunk 23]
```diff
@@ -3,7 +3,7 @@ import React from 'react';
 import { truncateMiddle } from '../../lib/textHelper';
 import { Loader } from '../UI';
 import AssetProxy, { createAssetProxy } from '../../valueObjects/AssetProxy';
-import styles from './FileControl.css';
+
```
Comment (Benaiah) [inline src/components/Widgets/ImageControl.js]: Remove extra blank lines from edited JS files.

File: src/components/Widgets/ListControl.js [hunk 24]
```diff
@@ -4,7 +4,7 @@ import { List, Map, fromJS } from 'immutable';
 import { sortable } from 'react-sortable';
 import FontIcon from 'react-toolbox/lib/font_icon';
 import ObjectControl from './ObjectControl';
-import styles from './ListControl.css';
+
```
Comment (Benaiah) [inline src/components/Widgets/ListControl.js]: Remove extra blank lines from edited JS files.

File: src/components/Widgets/Markdown/MarkdownControl/RawEditor/index.js [hunk 25]
```diff
@@ -5,7 +5,7 @@ import Plain from 'slate-plain-serializer';
 import { debounce } from 'lodash';
 import Toolbar from '../Toolbar/Toolbar';
 import { Sticky } from '../../../../UI/Sticky/Sticky';
-import styles from './index.css';
+
```
Comment (Benaiah) [inline src/components/Widgets/Markdown/MarkdownControl/RawEditor/index.js]: Remove extra blank lines from edited JS files.

File: src/components/Widgets/Markdown/MarkdownControl/Toolbar/Toolbar.js [hunk 26]
```diff
@@ -7,7 +7,7 @@ import ToolbarButton from './ToolbarButton';
 import ToolbarComponentsMenu from './ToolbarComponentsMenu';
 import ToolbarPluginForm from './ToolbarPluginForm';
 import { Icon } from '../../../../UI';
-import styles from './Toolbar.css';
+
```
Comment (Benaiah) [inline src/components/Widgets/Markdown/MarkdownControl/Toolbar/Toolbar.js]: Remove extra blank lines from edited JS files.

File: src/components/Widgets/Markdown/MarkdownControl/Toolbar/ToolbarButton.js [hunk 27]
```diff
@@ -2,11 +2,11 @@ import PropTypes from 'prop-types';
 import React from 'react';
 import classnames from 'classnames';
 import { Icon } from '../../../../UI';
-import styles from './ToolbarButton.css';
+
```
Comment (Benaiah) [inline src/components/Widgets/Markdown/MarkdownControl/Toolbar/ToolbarButton.js]: Remove extra blank lines from edited JS files.

File: src/components/Widgets/Markdown/MarkdownControl/Toolbar/ToolbarComponentsMenu.js [hunk 28]
```diff
@@ -3,7 +3,7 @@ import React from 'react';
 import ImmutablePropTypes from 'react-immutable-proptypes';
 import { Menu, MenuItem } from 'react-toolbox/lib/menu';
 import ToolbarButton from './ToolbarButton';
-import styles from './ToolbarComponentsMenu.css';
+
```
Comment (Benaiah) [inline src/components/Widgets/Markdown/MarkdownControl/Toolbar/ToolbarComponentsMenu.js]: Remove extra blank lines from edited JS files.

File: src/components/Widgets/Markdown/MarkdownControl/Toolbar/ToolbarPluginForm.js [hunk 29]
```diff
@@ -3,7 +3,7 @@ import React from 'react';
 import { Map } from 'immutable';
 import { Button } from 'react-toolbox/lib/button';
 import ToolbarPluginFormControl from './ToolbarPluginFormControl';
-import styles from './ToolbarPluginForm.css';
+
```
Comment (Benaiah) [inline src/components/Widgets/Markdown/MarkdownControl/Toolbar/ToolbarPluginForm.js]: Remove extra blank lines from edited JS files.

File: src/components/Widgets/Markdown/MarkdownControl/Toolbar/ToolbarPluginFormControl.js [hunk 30]
```diff
@@ -1,7 +1,7 @@
 import PropTypes from 'prop-types';
 import React from 'react';
 import { resolveWidget } from '../../../../Widgets';
-import styles from './ToolbarPluginFormControl.css';
+
```
Comment (Benaiah) [inline src/components/Widgets/Markdown/MarkdownControl/Toolbar/ToolbarPluginFormControl.js]: Remove extra blank lines from edited JS files.

File: src/components/Widgets/Markdown/MarkdownControl/VisualEditor/components.js [hunk 31]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import { List } from 'immutable';
 import cn from 'classnames';
-import styles from './index.css';
+
```
Comment (Benaiah) [inline src/components/Widgets/Markdown/MarkdownControl/VisualEditor/components.js]: Remove extra blank lines from edited JS files.

File: src/components/Widgets/Markdown/MarkdownControl/VisualEditor/index.js [hunk 32]
```diff
@@ -11,7 +11,7 @@ import { MARK_COMPONENTS, NODE_COMPONENTS } from './components';
 import RULES from './rules';
 import plugins, { EditListConfigured } from './plugins';
 import onKeyDown from './keys';
-import styles from './index.css';
+
```
Comment (Benaiah) [inline src/components/Widgets/Markdown/MarkdownControl/VisualEditor/index.js]: Remove extra blank lines from edited JS files.

File: src/components/Widgets/ObjectControl.js [hunk 33]
```diff
@@ -2,8 +2,7 @@ import PropTypes from 'prop-types';
 import React, { Component } from 'react';
 import { Map } from 'immutable';
 import { resolveWidget } from '../Widgets';
-import controlStyles from '../ControlPanel/ControlPane.css';
-import styles from './ObjectControl.css';
+
```
Comment (Benaiah) [inline src/components/Widgets/ObjectControl.js]: Remove extra blank lines from edited JS files.

File: src/containers/App.js [hunk 34]
```diff
@@ -21,8 +21,7 @@ import AppHeader from '../components/AppHeader/AppHeader';
 import { Loader, Toast } from '../components/UI/index';
 import { getCollectionUrl, getNewEntryUrl } from '../lib/urlHelper';
 import { SIMPLE, EDITORIAL_WORKFLOW } from '../constants/publishModes';
-import styles from './App.css';
-import sidebarStyles from './Sidebar.css';
+
```
Comment (Benaiah) [inline src/containers/App.js]: Remove extra blank lines from edited JS files.

File: src/containers/CollectionPage.js [hunk 35]
```diff
@@ -6,7 +6,7 @@ import { loadEntries } from '../actions/entries';
 import { selectEntries } from '../reducers';
 import { Loader } from '../components/UI';
 import EntryListing from '../components/EntryListing/EntryListing';
-import styles from "./CollectionPage.css";
+
```
Comment (Benaiah) [inline src/containers/CollectionPage.js]: Remove extra blank lines from edited JS files.

File: src/containers/Sidebar.js [hunk 36]
```diff
@@ -4,7 +4,7 @@ import { connect } from 'react-redux';
 import ReactSidebar from 'react-sidebar';
 import _ from 'lodash';
 import { openSidebar } from '../actions/globalUI';
-import styles from './Sidebar.css';
+
```
Comment (Benaiah) [inline src/containers/Sidebar.js]: Remove extra blank lines from edited JS files.

File: src/components/AppHeader/AppHeader.css [hunk 37]
```diff
@@ -1,6 +1,6 @@
-@import '../UI/theme';
+@import '../UI/theme.css';
```
Comment (erquhart) [inline src/components/AppHeader/AppHeader.css]: Don't import here

File: src/components/EntryEditor/EntryEditor.css [hunk 38]
```diff
@@ -1,39 +1,39 @@
-@import '../UI/theme';
+@import '../UI/theme.css';
```
Comment (erquhart) [inline src/components/EntryEditor/EntryEditor.css]: remove

File: src/components/EntryEditor/EntryEditor.js [hunk 39]
```diff
@@ -9,7 +9,6 @@ import ControlPane from '../ControlPanel/ControlPane';
 import PreviewPane from '../PreviewPane/PreviewPane';
 import Toolbar from './EntryEditorToolbar';
 import { StickyContext } from '../UI/Sticky/Sticky';
-import styles from './EntryEditor.css';
 import stickyStyles from '../UI/Sticky/Sticky.css';
```
Comment (erquhart) [inline src/components/EntryEditor/EntryEditor.js]: css import

File: src/backends/git-gateway/AuthenticationPage.css [hunk 40]
```diff
@@ -1,31 +1,31 @@
-.root {
+.nc-gitGatewayAuthenticationPage-root {
```
[the remark this answers is not in the document]
Comment (Benaiah) [reply src/backends/git-gateway/AuthenticationPage.css]: Agreed. Those were my thoughts as well, I just figured they were out of scope for this PR.

File: src/components/AppHeader/AppHeader.css (hunk 37 above)
Comment (Benaiah) [reply src/components/AppHeader/AppHeader.css]: Thanks, for some reason I had thought this was necessary to get the theme variables in this file. Reading the docs on it that's clearly not the case - not sure where I got that idea,

File: src/components/AppHeader/AppHeader.css [hunk 41]
```diff
@@ -9,18 +9,18 @@
...
 
-.homeLink,
-.iconMenu {
+.nc-appHeader-homeLink,
+.nc-appHeader-iconMenu {
   margin-left: 2%;
 }
 
-.homeLink .icon {
+.nc-appHeader-homeLink &icon {
```
[the remark this answers is not in the document]
Comment (Benaiah) [reply src/components/AppHeader/AppHeader.css]: :+1:

**Pull request merged**